### ISSUE TITLE
Field with the name of the request in k6 stats

### ISSUE THREAD

**Issue by daylikon** [mostafa/xk6-kafka#271]
Hello @mostafa!

At this moment, when generating a csv report and exporting metrics, you can see the metrics (for example, `kafka_writer_write_seconds`), the metric value, scenario name and to which topic message was sent in extra_tags for each request.

I want to somehow identify different requests that are sent to one topic in one scenario in stats, for example, using the `name`, `metadata`, `tags`/`extra_tags` fields. Is there a way to set these fields now?

If not, is it possible to issue this as a feature request?

Regards, Andrew

**Comment by mostafa** [comment]
Hey @daylikon,

k6 supports custom metrics: https://grafana.com/docs/k6/latest/using-k6/metrics/create-custom-metrics/
And with custom metrics, you can also have tags, like this counter metric example: https://grafana.com/docs/k6/latest/javascript-api/k6-metrics/counter/

**Comment by daylikon** [comment]
Yes, I can use k6 custom metrics, but I want to use xk6-kafka metric value (`kafka_writer_write_seconds`/`kafka_reader_read_seconds`/etc).

Right now writer returns `undefined`, the difference between the time calculated between writer.produce() is very different from the metric.

Is there a way to get plugin metrics after sending or reading a message, so that they can then be used in custom metrics?

**Comment by mostafa** [comment]
@daylikon 

I don't think so, unless you export them while the test is running.

**Comment by daylikon** [comment]
@mostafa

So I’m thinking about how to export metrics while the test is running. Messages are written in the same topic, but the “`name`” is different.

Now in the csv and given output plugin, if take the `kafka_writer_write_seconds` metric, I can compare by topic, but since the payload of the messages is different, I want to aggregate these messages by “`name`,” but now it’s impossible to do that.

**Comment by mostafa** [comment]
@daylikon 

Currently only static tags are added to [reader](https://github.com/mostafa/xk6-kafka/blob/5612337c573cfd0c915a9c6f0554b73fba3a72fa/reader.go#L399-L402) and [writer](https://github.com/mostafa/xk6-kafka/blob/5612337c573cfd0c915a9c6f0554b73fba3a72fa/writer.go#L275-L276) metrics based on the metrics provided from the kafka-go module's [readerStats](https://github.com/segmentio/kafka-go/blob/1b64d17c73e44e835841844213d35e252623fb6d/reader.go#L572-L604) and [writerStats](https://github.com/segmentio/kafka-go/blob/1b64d17c73e44e835841844213d35e252623fb6d/writer.go#L345-L387). You can modify the xk6-kafka's `reportReaderStats` and `reportWriterStats` functions to include dynamic tags and try to see if it works the way you expect.

**Comment by mostafa** [comment]
@daylikon 

Ref: https://github.com/mostafa/xk6-kafka/pull/272#issuecomment-1927305422
